File: app/src/main/java/com/savvasdalkitsis/gameframe/control/view/ControlView.java
package com.savvasdalkitsis.gameframe.control.view;

import com.savvasdalkitsis.gameframe.ip.model.IpAddress;

public interface ControlView {

    void operationSuccess();

    void operationFailure(Throwable e);

    void ipAddressLoaded(IpAddress ipAddress);

    void ipCouldNotBeFound(Throwable throwable);
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/control/view/ControlView.kt
package com.savvasdalkitsis.gameframe.control.view

import com.savvasdalkitsis.gameframe.ip.model.IpAddress

interface ControlView {

    fun operationSuccess()

    fun operationFailure(e: Throwable)

    fun ipAddressLoaded(ipAddress: IpAddress)

    fun ipCouldNotBeFound(throwable: Throwable)
}
